### ISSUE TITLE
UberBuffer index -1 must be between 0 and 1

### ISSUE THREAD

**Issue by james64** [lihaoyi/fastparse#208]
I am getting error in title. Working example below. This is probably related to #135 because both whitespaceApi and iterator parsing are involved. I use version 2.0.5

```
import fastparse._

object SortOfMinimalExample extends App {

  def p3[_:P] : P[Unit] = {
    import NoWhitespace._
    "a" ~ " ".rep ~/ "b"
  }

  def p2[_:P] : P[Unit] = {
    import CustomWhitespace._
    p3.rep(sep = "\n".rep)
  }

  def p1[_:P] : P[Unit] = {
    import NoWhitespace._
    p2 ~ "]"
  }

  parse("a b ]", p1(_)) // this works
  parse(Seq("a b ]").iterator, p1(_)) // this does not
}

object CustomWhitespace {
  import NoWhitespace._
  implicit val whitespace : P[_] => P[Unit] = { implicit ctx: P[_] => " " ~ "#".? }
}
```

**Comment by lihaoyi** [comment]
Slightly more minimized:
```scala
@ {
  import fastparse._

  def p3[_:P]: P[Unit] = {
    import NoWhitespace._
    "a" ~ " " ~/ "b"
  }

  def p2[_:P]: P[Unit] = {
    import CustomWhitespace._
    p3.rep() // Note: p3.rep passes!!!
  }

  def p1[_:P]: P[Unit] = {
    import NoWhitespace._
    p2 ~ "]"
  }

  object CustomWhitespace {
    import NoWhitespace._
    implicit val whitespace : P[_] => P[Unit] = { implicit ctx: P[_] => " " ~ " ".?}
  }
  parse(Seq("a b ]").iterator, p1(_)) // boom
  }
```

**Comment by lihaoyi** [comment]
This appears to be fixed in my latest dev branch https://github.com/lihaoyi/fastparse/tree/dynamic-aggregation. Unclear what fixed it, but there are a lot of fixes in that branch for various edge cases. I'll see if I can clean it up and publish

**Comment by lihaoyi** [comment]
Fixed in 2.1.0

**Comment by james64** [comment]
I confirm the fix works for me as well. Thanks for quick turnaround!
